rwslib/builders/admindata.py

```diff
@@ -0,0 +1,391 @@
+        self.locations = []
+        # SignatureDef
+
+    def build(self, builder):
+        """Build XML by appending to builder"""
+        params = {}
+        if self.study_oid:
+            params.update(dict(StudyOID=self.study_oid))
+        builder.start("AdminData", params)
+        for user in self.users:
+            user.build(builder)
+        for location in self.locations:
+            location.build(builder)
+
+        builder.end("AdminData")
+
+    def __lshift__(self, other):
+        """Override << operator"""
+
+        if not isinstance(other, (User, Location,)):
+            raise ValueError('{0} cannot accept a {1} as a child element'.format(self.__class__.__name__,
+                                                                                 other.__class__.__name__))
+
+        self.set_list_attribute(other, User, 'users')
+        self.set_list_attribute(other, Location, 'locations')
+
+        return other
+
+
+class MetaDataVersionRef(ODMElement):
+    """
+    A reference to a MetaDataVersion used at the containing Location. 
+      The EffectiveDate expresses the fact that the metadata used at a location can vary over time.
+    """
+    def __init__(self, study_oid, metadata_version_oid, effective_date):
+        """
+        :param str study_oid: References the :class:`Study` that uses this metadata version.
+        :param str metadata_version_oid: References the :class:`MetaDataVersion` (within the above Study).
+        :param datetime effective_date: Effective Date for this version and Site
+        """
+        super(MetaDataVersionRef, self).__init__()
+        self.study_oid = study_oid
+        self.metadata_version_oid = metadata_version_oid
+        self.effective_date = effective_date
+
+    def build(self, builder):
+        """Build XML by appending to builder"""
+        params = dict(StudyOID=self.study_oid,
+                      MetaDataVersionOID=self.metadata_version_oid,
+                      EffectiveDate=dt_to_iso8601(self.effective_date))
+        builder.start("MetaDataVersionRef", params)
+        builder.end("MetaDataVersionRef")
+
+
+class Location(ODMElement, LastUpdateMixin):
+    """
+    A physical location -- typically a clinical research site or a sponsor's office.
+    """
+    def __init__(self, oid, name,
+                 location_type=None,
+                 metadata_versions=[]):
+        """
+        :param str oid: OID for the Location, referenced in :class:`LocationRef`
+        :param str name: Name for the Location
+        :param rwslib.builder_constants.LocationType location_type: Type for this Location
+        :param list(MetaDataVersion) metadata_versions: The :class:`MetaDataVersionRef` for this Location
+        """
+        super(Location, self).__init__()
+        self.oid = oid
+        self.name = name
+        self._location_type = None
+        if location_type:
+            self.location_type = location_type
+        self.metadata_versions = []
```

Comment:
It's just confusing because you're expecting us to assign the metadata_versions passed to self.metadata_versions directly and we don't do that....

```python
       
    def __init__(self, oid, name,
                 location_type=None,
                 metadata_versions=[]):
   
       # We set it to a (new) empty list 
       self.metadata_versions = []
 
        # We "push" any metadata_versions we were sent into ourselves which has the side effect of checking they
        # actually are a metadataversion or other child type we can accept.
        for mdv in metadata_versions:
            self << mdv
```
Perhaps there is a better way to do this but since we never mutate the ```metadata_versions``` default argument we're not falling foul of the mutable argument trap. You could argue we're only a step away and I couldn't argue.